### PR TITLE
RK-17831 - feat: added support for pre-existing subnets

### PR DESCRIPTION
**Pull request by gosharo**

### CODE REVIEW

**Comment by gosharo** [comment]
Github Enforcer opened Task: [RK-17831](https://rookout.atlassian.net/browse/RK-17831)

[RK-17831]: https://rookout.atlassian.net/browse/RK-17831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: README.md [hunk 1]
```diff
@@ -127,13 +130,16 @@ No modules.
...
 | <a name="input_location"></a> [location](#input\_location) | Location of resources | `string` | `"northeurope"` | no |
 | <a name="input_private_endpoint_subnet_cidr"></a> [private\_endpoint\_subnet\_cidr](#input\_private\_endpoint\_subnet\_cidr) | CIDR of private endpoint, for internal deployment | `string` | `"10.10.0.64/26"` | no |
+| <a name="input_private_endpoint_subnet_name"></a> [private\_endpoint\_subnet\_name](#input\_private\_endpoint\_subnet\_name) | Private endpoints subnet, used for internal deployment, Minimum CIDR mask is 26 bits, should be in existing vnet resource group | `string` | `""` | no |
 | <a name="input_rookout_token"></a> [rookout\_token](#input\_rookout\_token) | Rookout's org token | `string` | n/a | yes |
 | <a name="input_subnet_app_serivce_cidr"></a> [subnet\_app\_serivce\_cidr](#input\_subnet\_app\_serivce\_cidr) | vnet subnets | `string` | `"10.10.0.0/26"` | no |
+| <a name="input_subnet_app_serivce_name"></a> [subnet\_app\_serivce\_name](#input\_subnet\_app\_serivce\_name) | App Service delegated subnet, Minimum CIDR mask is 26 bits, should be in existing vnet resource group | `string` | `""` | no |
```
Comment (idanrookout) [inline README.md]: service instead of serivce

File: examples/private_existing_vnet_and_subnets_existing_deployment_resource_group.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    rookout_token = "YOUR_TOKEN"
+    existing_resource_group_name = "..." # used for rookout deployment
+
+    internal = true
+
+    create_vnet = false
+    existing_vnet_name = "..."
+    existing_vnet_resource_group_name = "..."
+
+    subnet_app_serivce_name = "..."
```
Comment (idanrookout) [inline examples/private_existing_vnet_and_subnets_existing_deployment_resource_group.tf]: service

File: network.tf [hunk 3]
```diff
@@ -13,12 +13,22 @@ data "azurerm_virtual_network" "selected" {
...
+data "azurerm_subnet" "app_service_selected" {
+  count = !var.create_vnet && var.subnet_app_serivce_name == "" ? 0 : 1
+
+  name                 = var.subnet_app_serivce_name
+  virtual_network_name = data.azurerm_virtual_network.selected[0].name
+  resource_group_name  = var.existing_vnet_resource_group_name
 }
 
 resource "azurerm_subnet" "app_serivce" {
+  count = var.subnet_app_serivce_name == "" ? 1 : 0
```
Comment (idanrookout) [inline network.tf]: Change all serivce to service

**Review by idanrookout — CHANGES_REQUESTED** [review]

**Comment by idanrookout** [comment]
Also please provide description as expected by release please

**Comment by gosharo** [comment]
> Also please provide description as expected by release please

have added release-please label

**Review by idanrookout — APPROVED** [review]

**Pull request merged**